src/mc-fle2-payload-iev.c

```diff
@@ -200,6 +200,13 @@ mc_FLE2IndexedEncryptedValue_add_S_Key (_mongocrypt_crypto_t *crypto,
    /* Read ClientEncryptedValue. */
    uint64_t expected_length =
       mc_reader_get_consumed_length (&reader) + length - 16;
+   if (length > iev->Inner.len || expected_length > iev->Inner.len) {
+      CLIENT_ERR ("mc_FLE2IndexedEncryptedValue_add_S_Key expected "
+                  "Inner byte length >= %" PRIu32 " got: %" PRIu32,
+                  expected_length,
+                  iev->Inner.len);
+      return false;
+   }
```

Comment:
```suggestion
```

The failing logs suggest expected_length may need to be removed in addition to the if block:
```
[2022/12/16 21:49:48.667] /data/mci/d43c20c888394e4b5dce0ac4f164790f/libmongocrypt/src/mc-fle2-payload-iev.c:201:13: error: unused variable 'expected_length' [-Werror=unused-variable]
[2022/12/16 21:49:48.667]     uint64_t expected_length =
[2022/12/16 21:49:48.667]              ^~~~~~~~~~~~~~~
[2022/12/16 21:49:48.667] cc1: all warnings being treated as errors
```

Removing both seems preferable. But if it requires further investigation, this changes seems OK to get Evergreen green.